File: app/src/main/java/com/example/employeevolleyjava/MyApplication.java
package com.example.employeevolleyjava;

import android.app.Application;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.toolbox.Volley;

public class MyApplication extends Application {

    public static String TAG = MyApplication.class.getSimpleName();
    public static MyApplication instance = null;

    RequestQueue requestQueue = null;

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
    }

    public RequestQueue getRequestQueue() {
        if (requestQueue == null) {
            return Volley.newRequestQueue(getApplicationContext());
        } else
            return requestQueue;
    }


    public <T> void addToRequestQueue(Request<T> request){
        request.setTag(TAG);
        getRequestQueue().add(request);
    }


}

File: app/src/main/java/com/example/employeevolleykotlin/MyApplication.kt
package com.example.employeevolleykotlin

import android.app.Application
import com.android.volley.Request
import com.android.volley.RequestQueue
import com.android.volley.toolbox.Volley


class MyApplication : Application() {
    private var requestQueue: RequestQueue? = null
    override fun onCreate() {
        super.onCreate()
        instance = this
    }

    @JvmName("getRequestQueue1")
    fun getRequestQueue(): RequestQueue {
        return requestQueue ?: Volley.newRequestQueue(applicationContext)
    }

    fun <T> addToRequestQueue(request: Request<T>) {
        request.tag = TAG
        getRequestQueue().add(request)
    }

    companion object {
        var TAG = MyApplication::class.java.simpleName
        var instance: MyApplication? = null
    }
}